app/server.js

```diff
@@ -13,6 +16,36 @@ const users = require('../initial-data/users.json');
 const brands = require('../initial-data/brands.json');
 const products = require('../initial-data/products.json');
 
+
+// ApiKey Generator @ https://www.akto.io/tools/api-key-generator
+const VALID_APIKEY = ["kznylgunwckqhpmwofzgqetxzdaxafurgbbnfdcnjevfpkukshksujpfrvhanuftzudvwrmakpxdhznbecuuhckvdilihglyfqonetlnwtizmzbfztkqvpawnbdyvmni"];
+
+// CORS
+const CORS_HEADERS = {
+  "Access-Control-Allow-Origin": "*",
+  "Access-Control-Allow-Headers": "Origin, X-Requested-With, Content-Type, Accept, Authorization, X-Authentication, X-Username, X-Password, X-ApiKey",
+  "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS"
+};
+
+// Middleware stuff...
+app.use((req, res, next) => {
+  res.set(CORS_HEADERS);
+  if (req.method === 'OPTIONS') {
+    return res.status(200).end();
+  }
+  next();
+});
+
+// Authentication middleware
+const authenticate = (req, res, next) =>  {
```

Comment:
this is not really authenticating. 
You need to basically  accessToken that was returned in the /login api. 
The login api needs to check the sent user & password and then generate that token that is checked on each api